web/core/management/commands/ensure_db.py

```diff
@@ -155,15 +183,14 @@ def _ensure_users_and_db(self, admin_conn: Connection):
 
                 # Ensure DB User Exists
                 if not self._user_exists(cursor, db_user):
-                    admin_user = admin_conn.info.user
-                    self._create_database_user(cursor, admin_user, db_alias, db_user, db_password)
-                    self._ensure_schema_permissions(db_name, db_user)
+                    self._create_database_user(cursor, admin_conn.info.user, db_alias, db_user, db_password)
                 else:
                     self.stdout.write(f"User found: {db_user}")
 
                 # Ensure Database Exists
                 if not self._database_exists(cursor, db_name):
                     self._create_database(cursor, db_alias, db_name, db_user)  # db_user is the owner
+                    self._ensure_schema_permissions(db_name, db_user)
```

Comment:
This is the main fix in this PR, moving the call to `self._ensure_schema_permission` into the database creation block.